### PR TITLE
fix(install): fix timezone getter in unattended script

### PR DESCRIPTION
**Pull request by kduret**
## Description

Fix unattended.sh when timezone name is not compatible with php

**Fixes** MON-5376

## Type of change

- [x] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [ ] 19.04.x
- [ ] 19.10.x
- [x] 20.04.x
- [x] 20.10.x (master)

<h2> How this pull request can be tested ? </h2>

Run unattended.sh with specific timezones

### CODE REVIEW

**Comment by Nohzoh** [comment]
wouldn't it be interesting to accept a TZ environment variable in order to force the timezone ?
then we could imagine that the download web site provide explicitly the user timezone in the command line provided to execute the unattended script

**Comment by kduret** [comment]
> wouldn't it be interesting to accept a TZ environment variable in order to force the timezone ?
> then we could imagine that the download web site provide explicitly the user timezone in the command line provided to execute the unattended script

Yes we could, but I think it's more interesting to do this automatically, cause I guess most f the users want to synchronize the centreon timezone with the server timezone.
But indeed, we could add the possibility to add an argument for that

**Review by jeremyjaouen — APPROVED** [review]

File: unattended.sh [hunk 1]
```diff
@@ -114,10 +114,19 @@ print_step_end
...
+        $timezoneName = $matches[1];
+    }
+
+    if (date_default_timezone_set($timezoneName) === false) {
+      $timezoneName = "UTC";
+    }
+
+    echo $timezoneName;
+' 2> /dev/null`
 echo "date.timezone = $timezone" > /etc/opt/rh/rh-php72/php.d/10-centreon.ini
```
Comment (sc979) [inline unattended.sh:130]: Caution : this file may not exists : 
https://docs.centreon.com/current/en/upgrade/upgrade-from-19-04.html#update-the-php-version

Comment (Nohzoh) [reply unattended.sh:130]: I think that this script is done to make an install from scratch.
It should not be executed on a machine that already has a Centreon installed.

If this requirement is not met, many things could fail while executing this script.

**Review by sc979 — APPROVED** [review]

**Comment by cgagnaire** [comment]
Closes https://github.com/centreon/centreon/issues/8868

**Pull request merged**